src/dex-ui/pages/ProposalDetails/ProposalDetails.tsx

```diff
@@ -43,7 +43,7 @@ export const ProposalDetails = () => {
 
   const castVote = (voteType: VoteType) => {
     if (formattedProposal) {
-      governance.castVote(formattedProposal.id, voteType);
+      governance.castVote(formattedProposal.contractId, formattedProposal.id, voteType);
```

Comment:
We need the proposals contractId to cast the vote on the correct proposal contract.